limbo/plugins/weather.py

```diff
@@ -5,59 +5,142 @@
     from urllib import quote
 except ImportError:
     from urllib.request import quote
+import json
 import logging
 import os
 import re
 import time
 
 import requests
 
-# http://openweathermap.org/weather-conditions
-iconmap = {
-    "01": ":sunny:",
-    "02": ":partly_sunny:",
-    "03": ":partly_sunny:",
-    "04": ":cloud:",
-    "09": ":droplet:",
-    "10": ":droplet:",
-    "11": ":zap:",
-    "13": ":snowflake:",
-    "50": ":umbrella:",    # mist?
+# https://developer.yahoo.com/weather/documentation.html
+ICONMAP = {
+    "0": ":tornado:",
+    "1": ":cyclone:",
+    "2": ":cyclone:",
+    "3": ":thunder_cloud_and_rain:",
+    "4": ":thunder_cloud_and_rain:",
+    "5": ":snowflake:",
+    "6": ":snowflake:",
+    "7": ":snowflake:",
+    "8": ":snowflake:",
+    "9": ":rain_cloud:",
+    "10": ":snowflake:",
+    "11": ":rain_cloud:",
+    "12": ":rain_cloud:",
+    "13": ":snowflake",
+    "14": ":snowflake:",
+    "15": ":snowflake:",
+    "16": ":snowflake:",
+    "17": ":thunder_cloud_and_rain:",
+    "18": ":rain_cloud:",
+    "19": ":desert:",
+    "20": ":cloud:",
+    "21": ":cloud:",
+    "22": ":smoking:",
+    "23": ":wind_blowing_face:",
+    "24": ":wind_blowing_face:",
+    "25": ":snowman_without_snow:",
+    "26": ":cloud:",
+    "27": ":sun_behind_cloud:",
+    "28": ":sun_behind_cloud:",
+    "29": ":sun_small_cloud:",
+    "30": ":sun_small_cloud:",
+    "32": ":sunny:",
+    "33": ":sunny:",
+    "34": ":sunny:",
+    "35": ":thunder_cloud_and_rain:",
+    "36": ":sunny:",
+    "37": ":thunder_cloud_and_rain:",
+    "38": ":thunder_cloud_and_rain:",
+    "39": ":thunder_cloud_and_rain:",
+    "40": ":rain_cloud:",
+    "41": ":snowflake:",
+    "42": ":snowflake:",
+    "43": ":snowflake:",
+    "44": ":partly_sunny:",
+    "45": ":thunder_cloud_and_rain:",
+    "46": ":snowflake:",
+    "47": ":thunder_cloud_and_rain:",
 }
 
+class WeatherException(Exception):
+    """An exception finding the weather"""
+    pass
+
 def weather(searchterm):
-    weather_api_key = os.environ.get("WEATHER_API_KEY")
-    if not weather_api_key:
-        return "Please set as the WEATHER_API_KEY environment variable to a " \
-               "valid (free) OpenWeatherMap API key: " \
-               "http://openweathermap.org/appid#get"
+    """Get the weather for a place given by searchterm
+
+    Returns a title and a list of forecasts.
+
+    The title describes the location for the forecast (i.e. "Portland, ME USA")
+    The list of forecasts is a list of dictionaries in slack attachment fields
+        format (see https://api.slack.com/docs/message-attachments)
+
+    Throws WeatherException if the location given by `searchterm` can't be
+    found.
+    """
+    yql = u'select * from weather.forecast where woeid in '\
+          '(select woeid from geo.places(1) where text="{}")'.format(
+              searchterm)
+
+    unit = "c" if os.environ.get("WEATHER_CELSIUS") else "f"
```

Comment:
Maybe would be cleaner to pass the server and do ```server.config.get("WEATHER_CELSIUS")```?

Reply:
I think it will make sense to unify access to config variables, but I'll do that as part of merging your PR 👍